var/spack/repos/builtin/packages/r-acde/package.py

```diff
@@ -7,21 +7,27 @@
 
 
 class RAcde(RPackage):
-    """This package provides a multivariate inferential analysis method
-    for detecting differentially expressed genes in gene expression data.
-    It uses artificial components, close to the data's principal
-    components but with an exact interpretation in terms of differential
-    genetic expression, to identify differentially expressed genes while
-    controlling the false discovery rate (FDR). The methods on this
-    package are described in the vignette or in the article
-    'Multivariate Method for Inferential Identification of
-    Differentially Expressed Genes in Gene Expression Experiments' by
-    J. P. Acosta, L. Lopez-Kleine and S. Restrepo
-    (2015, pending publication)."""
+    """Artificial Components Detection of Differentially Expressed Genes.
 
-    homepage = "https://www.bioconductor.org/packages/acde/"
-    url      = "https://bioconductor.org/packages/release/bioc/src/contrib/acde_1.14.0.tar.gz"
+       This package provides a multivariate inferential analysis method for
+       detecting differentially expressed genes in gene expression data. It
+       uses artificial components, close to the data's principal components but
+       with an exact interpretation in terms of differential genetic
+       expression, to identify differentially expressed genes while controlling
+       the false discovery rate (FDR). The methods on this package are
+       described in the vignette or in the article 'Multivariate Method for
+       Inferential Identification of Differentially Expressed Genes in Gene
+       Expression Experiments' by J. P. Acosta, L. Lopez-Kleine and S. Restrepo
+       (2015, pending publication)."""
 
-    version('1.14.0', sha256='e8d9ad688610d4b72cab729141dc46fce580c97128d929885336bb138345b788')
+    homepage = "https://bioconductor.org/packages/acde"
+    git      = "https://git.bioconductor.org/packages/acde.git"
 
-    depends_on('r-boot', type=('build', 'run'))
+    version('1.14.0', commit='6017c7436a46f186b2a3cea9d2b93274f6dd3417')
+    version('1.12.0', commit='f6ce5926ac915c2d73436f47daf7f9791645dad4')
+    version('1.10.0', commit='2c303dec45f3c70bf333a6eacae568a08d5ca010')
+    version('1.8.0', commit='f7fc3e1dce958445f920d3b28b56abde70bfb9de')
+    version('1.6.0', commit='244c81f435a077bf7895ea565fa6695e8b079f67')
+
+    depends_on('r-boot@1.3:', when='@1.6.0:', type=('build', 'run'))
+    depends_on('r@3.3:', when='@1.6.0:', type=('build', 'run'))
```

Comment:
I prefer to order dependencies in the same order they appear in the description file, just so it's easier to check if they are all present. I also put R first in the list. Not required, but would make this easier to review if it isn't much work to change in your script.

Reply:
Updated -- the dependencies are now ordered as they appear in the description.